src/qtism/runtime/storage/binary/AbstractQtiBinaryStorage.php

```diff
@@ -340,7 +340,31 @@ public function retrieve($sessionId)
                     $durationStore->setVariable($durationVariable);
                 }
 
-                $assessmentTestSession->setDurationStore($durationStore);
+            $item = $assessmentTestSession->getCurrentAssessmentItemRef();
```

Comment:
I'm not sure about that approach.
The purpose of this method is to read and unmarshal data as-is. If some data is lost during marshaling, it should then probably be fixed there.

Reply:
The problem is that it does marshaling by inner data of item, in that case, it skips to read time limits and it not included it to duration storage, so this block is cover that missing

Reply:
That's all right. I mean, you have already replaced the data-storage usage with currentAssessmentItemRef's max time in timer-init time.

Let's revert these changes.